tests/approaches/test_grammar_search_invention_approach.py

```diff
@@ -126,9 +121,11 @@ def test_halving_constant_generator():
     """Tests for _halving_constant_generator().
     """
     expected_sequence = [0.5, 0.25, 0.75, 0.125, 0.625, 0.375, 0.875]
+    expected_costs = [1.0, 2.0, 2.0, 3.0, 3.0, 3.0, 3.0]
     generator = _halving_constant_generator(0., 1.)
-    for i, x in zip(range(len(expected_sequence)), generator):
-        assert abs(expected_sequence[i] - x) < 1e-6
+    for i, (constant, cost) in zip(range(len(expected_sequence)), generator):
```

Comment:
You can zip 3 lists in python:
`for (expected_constant, expected_cost, (constant, cost)) in zip(expected_sequence, expected_costs, generator):`